modulus/distributed/mappings.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+from modulus.distributed.manager import DistributedManager
+from modulus.distributed.utils import _reduce
+from modulus.distributed.utils import _split
+from modulus.distributed.utils import _gather
+
+
+# matmul parallel
+class _CopyToMatmulParallelRegion(torch.autograd.Function):
```

Comment:
To make this primitives more generic and potentially re-usable, I'd propose
- adding the name of the used process group as additional input to `forward`
- dropping explicit mentions of "MatmulParallel" or "SpatialParallel" in favor of having generic autograd functions like `distributed_tensor_split`, `distributed_tensor_all_reduce`, `distributed_tensor_all_gather` etc.

Reply:
Yes, I agree with this as well. In fact, this was why I had these utilities in `modulus/models/afno/distributed` before, but it makes sense to generalize them.

Reply:
Made this change to remove all spatial and matmul parallel specific references